libsolidity/formal/SMTChecker.cpp

```diff
@@ -62,6 +62,8 @@ void SMTChecker::endVisit(VariableDeclaration const& _varDecl)
 {
 	if (_varDecl.isLocalVariable() && _varDecl.type()->isValueType() &&_varDecl.value())
 		assignment(_varDecl, *_varDecl.value(), _varDecl.location());
+	else if (_varDecl.isStateVariable() && _varDecl.type()->isValueType())
```

Comment:
This probably fails if variables are declared below functions. I think we have to manually register them.

Reply:
True.. I read somewhere that state vars were visited before functions, but you're right.